modules/osdk-multi-arch-validate.adoc

```diff
@@ -0,0 +1,31 @@
+// Module included in the following assemblies:
+//
+// * operators/operator_sdk/osdk-multi-arch-support.adoc
+
+:_content-type: PROCEDURE
+[id="osdk-multi-arch-validate_{context}"]
+= Validating your Operator's multi-platform readiness
+
+You can use the Operator SDK `bundle validate` command to verify that the images defined in the cluster service version (CSV) deployment spec comply with the best practices as defined by the Operator Framework community.
+
+.Procedure
+
+* Run the following command to validate your Operator project for multiple architecture readiness:
++
+[source,terminal]
+----
+$ operator-sdk bundle validate ./bundle \
+  --select-optional name=multiarch
+----
++
+.Example validation message
+[source,text]
+----
+Some output
+----
++
+.Example error message
+[source,text]
+----
+Some output
```

Comment:
If your manager image is missing CSV labels:
```
ERRO[0016] Error: Value test-operator.v0.0.1: not all images specified are providing the support described via the CSV labels. Note that (SO.architecture): (linux.ppc64le) was not found for the image(s) [quay.io/jpoulin/test-operator:v1alpha1] 
ERRO[0016] Error: Value test-operator.v0.0.1: not all images specified are providing the support described via the CSV labels. Note that (SO.architecture): (linux.s390x) was not found for the image(s) [quay.io/jpoulin/test-operator:v1alpha1] 
ERRO[0016] Error: Value test-operator.v0.0.1: not all images specified are providing the support described via the CSV labels. Note that (SO.architecture): (linux.amd64) was not found for the image(s) [quay.io/jpoulin/test-operator:v1alpha1] 
ERRO[0016] Error: Value test-operator.v0.0.1: not all images specified are providing the support described via the CSV labels. Note that (SO.architecture): (linux.arm64) was not found for the image(s) [quay.io/jpoulin/test-operator:v1alpha1] 
```

If your manager image is healthy:
```
INFO[0020] All validation tests have completed successfully 
```

Reply:
All of that said, it appears I forgot to bump the version of operator-framework/api used by operator-sdk. This means that the validate provided out of the box won't scan for mismatched affinity configurations. We may want to make that another follow-up task so that we can update the docs with the output that includes more information about affinity.

Reply:
Here is the current output if your manager is successfully manifest-listed but you forgot the operatorhub flags:
```
WARN[0014] Warning: Value test-operator.v0.0.1: check if the CSV is missing the label (operatorframework.io/arch.<value>) for the Arch(s): ["amd64" "arm64" "ppc64le" "s390x"]. Be aware that your Operator manager image ["quay.io/jpoulin/test-operator:v1alpha1"] provides this support. Thus, it is very likely that you want to provide it and if you support more than amd64 architectures, you MUST,use the required labels for all which are supported.Otherwise, your solution cannot be listed on the cluster for these architectures 
```

Reply:
Output examples added. @jaypoulz Do you think we need to add an admonition stating that the `multiarch` option does not validate for node affinity?

Or just fix it soon and update?

Reply:
I think it's an enhancement moreso than an omission.
That said, I'll submit the PR today, and it will come in v1.32.0 of Operator SDK at the soonest.

Reply:
So I would probably provide the user with the info we have released today, and then follow up with the updates in a Z-stream when the new version of Operator SDK is released.

Reply:
Addressed in d0e842feaeba23ceeec245e52a74cba537e282b0

WDYT?

Reply:
New commit (https://github.com/openshift/openshift-docs/commit/d0e842feaeba23ceeec245e52a74cba537e282b0) makes sense. A few minor notes:
1. You used the British spelling of labelled (IDK if that matters) :)
2. IDK if it's appropriate to mention that a future update to operator SDK will also include checks for node affinity. I think probably we leave that bit for the z-stream fixup because it will also touch up the success / error output sections.

Reply:
We try not to talk about the future in docs. It leads to trouble more often than not. :)

And oops on the British spelling! Should be fixed now.

Reply:
Excellent. :) The PR above is already merged, but it makes no sense to mention a release (v1.32.0) of operator SDK when it doesn't exist yet. I think all my comments/insights are resolved at this point.